librz/include/rz_type.h

```diff
@@ -95,22 +101,32 @@ typedef struct rz_base_type_t {
 // AST-level types for C and C++
 // Parses strings like "const char * [0x42] const * [23]" to RzType
 
-typedef struct rz_ast_parser_t RzASTParser;
-
 typedef enum {
 	RZ_TYPE_KIND_IDENTIFIER,
 	RZ_TYPE_KIND_POINTER,
-	RZ_TYPE_KIND_ARRAY
+	RZ_TYPE_KIND_ARRAY,
+	RZ_TYPE_KIND_CALLABLE
 } RzTypeKind;
 
 typedef enum {
-	RZ_TYPE_CTYPE_IDENTIFIER_KIND_UNSPECIFIED,
-	RZ_TYPE_CTYPE_IDENTIFIER_KIND_STRUCT,
-	RZ_TYPE_CTYPE_IDENTIFIER_KIND_UNION,
-	RZ_TYPE_CTYPE_IDENTIFIER_KIND_ENUM
+	RZ_TYPE_IDENTIFIER_KIND_UNSPECIFIED,
+	RZ_TYPE_IDENTIFIER_KIND_STRUCT,
+	RZ_TYPE_IDENTIFIER_KIND_UNION,
+	RZ_TYPE_IDENTIFIER_KIND_ENUM
 } RzTypeIdentifierKind;
 
-typedef struct rz_type_t RzType;
+typedef struct rz_callable_arg_t {
+	RZ_NULLABLE char *name; // optional
+	RzType *type;
+} RzCallableArg;
+
+typedef struct rz_callable_at {
+	const char *name;
```

Comment:
Can we make this optional too?

Reply:
Making this optional in general is fine. But for saving the type in the `RzTypeDB->callables` hashtable we require a name.

Reply:
Yeah, I think we should make the name required for these exact functions and document it, but in the structure have it optional.

Reply:
I made it optional.